packages/gradient-block/src/GradientBlock.tsx

```diff
@@ -0,0 +1,150 @@
+/* (c) Copyright Frontify Ltd., all rights reserved. */
+
+import { MouseEvent, useEffect, useRef, useState } from 'react';
+import '@frontify/fondue-tokens/styles';
+import { useBlockSettings, useEditorState } from '@frontify/app-bridge';
+import { BlockProps } from '@frontify/guideline-blocks-settings';
+import { Divider } from '@frontify/fondue';
+import 'tailwindcss/tailwind.css';
+import { GradientColor, Settings, gradientHeightValues, gradientOrientationValues } from './types';
+import { DEFAULT_GRADIENT_COLORS, DEFAULT_HEIGHT_VALUE, DEFAULT_ORIENTATION_VALUE } from './constants';
+import { AddColorButton, ColorPicker, ColorTooltip, CssValueDisplay, SquareBadgesRow } from './components';
+import { toHexString } from '@frontify/guideline-blocks-shared';
+
+export const GradientBlock = ({ appBridge }: BlockProps) => {
+    const [blockSettings, setBlockSettings] = useBlockSettings<Settings>(appBridge);
+    const isEditing = useEditorState(appBridge);
+    const {
+        gradientColors,
+        isOrientationCustom,
+        orientationCustom,
+        orientationSimple,
+        isHeightCustom,
+        heightCustom,
+        heightSimple,
+        displayCss,
+    } = blockSettings;
+    const gradientBlockRef = useRef<HTMLDivElement>(null);
+    const [addButtonPositionLeft, setAddButtonPositionLeft] = useState(0);
+    const [currentlyEditingPosition, setCurrentlyEditingPosition] = useState(0);
+    const [showAddButton, setShowAddButton] = useState(false);
+    const [showColorModal, setShowColorModal] = useState(false);
+    const [isMounted, setMounted] = useState(false);
+
+    if (!gradientColors) {
+        setBlockSettings({
+            gradientColors: DEFAULT_GRADIENT_COLORS,
+        });
+    }
+
+    const setGradientColors = (colors: GradientColor[]) => {
+        setBlockSettings({
+            gradientColors: colors,
+        });
+    };
+
+    const gradientOrientation = isOrientationCustom
+        ? orientationCustom
+        : gradientOrientationValues[orientationSimple ?? DEFAULT_ORIENTATION_VALUE];
+
+    const gradientBlockHeight = isHeightCustom
+        ? heightCustom
+        : gradientHeightValues[heightSimple ?? DEFAULT_HEIGHT_VALUE];
+
+    const parseGradientColorsToCss = () => {
```

Comment:
Maybe move this also to the `helpers` and use it by passing in the color values. You could also add tests for it then.

Reply:
moved some parts of it into the helpers and added tests for it

Reply:
I am sorry. 

added a new helper `cssValueHelper`.

Unfortunately i wasn't able to add tests, i always got a lodash error (see screenshot)

![Screenshot 2023-03-13 at 15 39 31](https://user-images.githubusercontent.com/9362990/224735605-a9b0558c-0e9e-4e57-8705-718f95d74a5e.png)

The error is somehow coming from the `toHexString` function within shared as soon i remove this from the helper the test suite stops failing